# -*- coding: UTF-8 -*-
"""
FileName: serialwindow.py
Author: jiaminbit@sina.com
Create date: 2017.6.20
description: 串口调试工具UI
Update date：2017.7.20
version 1.0.0
"""

from PyQt5 import QtCore, QtGui, QtWidgets

class Ui_serialtool(object):
    def setupUi(self, serialtool):
        serialtool.setObjectName("serialtool")
        serialtool.resize(564, 362)
        self.gridLayout = QtWidgets.QGridLayout(serialtool)
        self.gridLayout.setObjectName("gridLayout")
        self.cb_serialname = QtWidgets.QComboBox(serialtool)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.cb_serialname.sizePolicy().hasHeightForWidth())
        self.cb_serialname.setSizePolicy(sizePolicy)
        self.cb_serialname.setObjectName("cb_serialname")
        self.gridLayout.addWidget(self.cb_serialname, 1, 0, 1, 1)
        self.te_serialrecvmsg = QtWidgets.QTextEdit(serialtool)
        self.te_serialrecvmsg.setObjectName("te_serialrecvmsg")
        self.gridLayout.addWidget(self.te_serialrecvmsg, 1, 1, 6, 1)
        self.cb_baund = QtWidgets.QComboBox(serialtool)
        self.cb_baund.setObjectName("cb_baund")
        self.cb_baund.addItem("")
        self.cb_baund.addItem("")
        self.cb_baund.addItem("")
        self.cb_baund.addItem("")
        self.cb_baund.addItem("")
        self.gridLayout.addWidget(self.cb_baund, 2, 0, 1, 1)
        self.cb_databit = QtWidgets.QComboBox(serialtool)
        self.cb_databit.setObjectName("cb_databit")
        self.cb_databit.addItem("")
        self.cb_databit.addItem("")
        self.cb_databit.addItem("")
        self.cb_databit.addItem("")
        self.gridLayout.addWidget(self.cb_databit, 3, 0, 1, 1)
        self.cb_checkbit = QtWidgets.QComboBox(serialtool)
        self.cb_checkbit.setObjectName("cb_checkbit")
        self.cb_checkbit.addItem("")
        self.cb_checkbit.addItem("")
        self.cb_checkbit.addItem("")
        self.cb_checkbit.addItem("")
        self.cb_checkbit.addItem("")
        self.gridLayout.addWidget(self.cb_checkbit, 4, 0, 1, 1)
        self.cb_stopbit = QtWidgets.QComboBox(serialtool)
        self.cb_stopbit.setObjectName("cb_stopbit")
        self.cb_stopbit.addItem("")
        self.cb_stopbit.addItem("")
        self.cb_stopbit.addItem("")
        self.gridLayout.addWidget(self.cb_stopbit, 5, 0, 1, 1)
        self.cb_fluid = QtWidgets.QComboBox(serialtool)
        self.cb_fluid.setObjectName("cb_fluid")
        self.cb_fluid.addItem("")
        self.cb_fluid.addItem("")
        self.cb_fluid.addItem("")
        self.gridLayout.addWidget(self.cb_fluid, 6, 0, 1, 1)
        self.pb_serialcon = QtWidgets.QPushButton(serialtool)
        self.pb_serialcon.setObjectName("pb_serialcon")
        self.gridLayout.addWidget(self.pb_serialcon, 7, 0, 1, 1)
        self.te_serialsendmsg = QtWidgets.QTextEdit(serialtool)
        self.te_serialsendmsg.setObjectName("te_serialsendmsg")
        self.gridLayout.addWidget(self.te_serialsendmsg, 7, 1, 2, 1)
        self.pb_serialsend = QtWidgets.QPushButton(serialtool)
        self.pb_serialsend.setObjectName("pb_serialsend")
        self.gridLayout.addWidget(self.pb_serialsend, 8, 0, 1, 1)
        self.lb_serialtitle = QtWidgets.QLabel(serialtool)
        font = QtGui.QFont()
        font.setPointSize(20)
        self.lb_serialtitle.setFont(font)
        self.lb_serialtitle.setAlignment(QtCore.Qt.AlignCenter)
        self.lb_serialtitle.setObjectName("lb_serialtitle")
        self.gridLayout.addWidget(self.lb_serialtitle, 0, 0, 1, 2)

        self.retranslateUi(serialtool)
        QtCore.QMetaObject.connectSlotsByName(serialtool)

    def retranslateUi(self, serialtool):
        _translate = QtCore.QCoreApplication.translate
        serialtool.setWindowTitle(_translate("serialtool", "Dialog"))
        self.cb_baund.setItemText(0, _translate("serialtool", "9600"))
        self.cb_baund.setItemText(1, _translate("serialtool", "19200"))
        self.cb_baund.setItemText(2, _translate("serialtool", "38400"))
        self.cb_baund.setItemText(3, _translate("serialtool", "57600"))
        self.cb_baund.setItemText(4, _translate("serialtool", "115200"))
        self.cb_databit.setItemText(0, _translate("serialtool", "8"))
        self.cb_databit.setItemText(1, _translate("serialtool", "7"))
        self.cb_databit.setItemText(2, _translate("serialtool", "6"))
        self.cb_databit.setItemText(3, _translate("serialtool", "5"))
        self.cb_checkbit.setItemText(0, _translate("serialtool", "None"))
        self.cb_checkbit.setItemText(1, _translate("serialtool", "Even"))
        self.cb_checkbit.setItemText(2, _translate("serialtool", "Odd"))
        self.cb_checkbit.setItemText(3, _translate("serialtool", "Mark"))
        self.cb_checkbit.setItemText(4, _translate("serialtool", "Space"))
        self.cb_stopbit.setItemText(0, _translate("serialtool", "1"))
        self.cb_stopbit.setItemText(1, _translate("serialtool", "1.5"))
        self.cb_stopbit.setItemText(2, _translate("serialtool", "2"))
        self.cb_fluid.setItemText(0, _translate("serialtool", "None"))
        self.cb_fluid.setItemText(1, _translate("serialtool", "CTS/RTS"))
        self.cb_fluid.setItemText(2, _translate("serialtool", "XON/XOFF"))
        self.pb_serialcon.setText(_translate("serialtool", "Connect"))
        self.pb_serialsend.setText(_translate("serialtool", "Send"))
        self.lb_serialtitle.setText(_translate("serialtool", "Serial Debug Tool"))

